### ISSUE TITLE
Identify/document how to use Experiments with SPAs

### ISSUE THREAD

**Issue by john-thomas-dotcms** [dotCMS/core#26308]
Experiments is currently designed and tested only for use in the dotCMS BE.  Customers will want to use Experiments with EMA and their SPAs.

We need to create a sample SPA which uses EMA, to demonstrate how an SPA can interface with Experiments via the API.

**Comment by oidacra** [comment]
P.O.C.
A page ( `/sale`) was created to generate an experiment with 1 variant.
Used: NextJS 14 (js) in "Client Side Rendering (CSR)" way (logic running in the browser/client)

Discoveries:
- We can run the Experiment and switch between variants depending on the assigned experiment. Unlike what was done before, the JS logic was centralized in obtaining the experiments and manipulating the localstorage which is the persistent source of the experiments. Tracking detect and listen events are handle direct in a Component in NextJS.
- I recommend migrating the logic to a library, to centralize and reuse in the various ways an experiment can be run (dotcms default, Sever Side Generated (SSG), Static Site Rendered (SSR), Static generation with dotCMS)
- Jitsu was used directly with the SDK.
- Instead of doing redirect, I filtered the 'a' with the href that matched the `pathurl` and to these I added the `variantName` queryparam and triggered the Jitsu event manually (in my case I didn't do it like that, Jitsu has a method to register the pageview automatically), this opens the door so that if we have defined in other goals such as `Click On Element`, being able to add a listener to those specific elements and trigger events from that element to Jitsu.
- It is necessary to verify the extra cases to take into account in javascript so that it works 100% the same as its current state in DotCMS.

Page/Experiment
Default
<img width="1605" alt="Screenshot 2024-02-05 at 11 59 54 AM" src="https://github.com/dotCMS/core/assets/1909643/72edb4ee-6478-460a-b739-c6ffdacbe187">

Variant
<img width="1606" alt="Screenshot 2024-02-05 at 11 59 44 AM" src="https://github.com/dotCMS/core/assets/1909643/586630b1-2119-4287-9874-26dbf9339edf">

Experiment and Variant Switch
<img width="3008" alt="Screenshot 2024-02-05 at 11 59 02 AM" src="https://github.com/dotCMS/core/assets/1909643/9652cdb0-ddc9-43d4-b904-323ed4901a61">

`PageView` event dispatched
<img width="1489" alt="Screenshot 2024-02-05 at 11 59 22 AM" src="https://github.com/dotCMS/core/assets/1909643/55c7baaa-747d-4b90-aeb5-8f5e90fe926c">
